### PR TITLE
fix parse annotation locale tr_TR

### PR DESCRIPTION
**Pull request by Defender32**
Issue https://github.com/doctrine/annotations/issues/265
Class DocLexer not correct parsing multi byte string

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php [hunk 1]
```diff
@@ -308,4 +308,26 @@ public function testTokenAdjacency() : void
         self::assertFalse($lexer->nextTokenIsAdjacent());
         self::assertFalse($lexer->moveNext());
     }
+
+    public function testMarkerAnnotationLocaleTr() : void
+    {
+        setlocale(LC_ALL, 'tr_TR.utf8', 'tr_TR');
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: Are we resetting the locale after this test?

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 2]
```diff
@@ -1398,6 +1398,14 @@ public function testWillParseAnnotationSucceededByANonImmediateDash()
         self::assertCount(1, $result);
         self::assertInstanceOf(SomeAnnotationClassNameWithoutConstructorAndProperties::class, $result[0]);
     }
+
+    public function testMultiByteAnnotationLocaleTR()
+    {
+        setlocale(LC_ALL, 'tr_TR.utf8', 'tr_TR');
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Are we resetting the locale after this test?

File: lib/Doctrine/Common/Annotations/DocLexer.php [hunk 3]
```diff
@@ -109,6 +109,14 @@ protected function getNonCatchablePatterns()
         return ['\s+', '\*+', '(.)'];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function getModifiers()
+    {
+        return parent::getModifiers() . 'u';
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocLexer.php]: Should this maybe be solved upfront in the abstract lexer?

Comment (Defender32) [reply lib/Doctrine/Common/Annotations/DocLexer.php]: I'll do it in an abstract lexer

Comment (Defender32) [reply lib/Doctrine/Common/Annotations/DocLexer.php]: create PR https://github.com/doctrine/lexer/pull/35

**Comment by greg0ire** [comment]
Please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches).

In your case, try avoid using "fix" without saying exactly what was broken, and why your changes improve the situation.

File: tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php (hunk 1 above)
Comment (Defender32) [reply tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: Add reset locale after this test

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php (hunk 2 above)
Comment (Defender32) [reply tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Add reset locale after this test

File: tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php (hunk 1 above)
Comment (alcaeus) [reply tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: Either reset the locale in a `tearDown` method or use `finally`: a failing assertion will leave the locale changed to `tr_TR`.

Comment (Defender32) [reply tests/Doctrine/Tests/Common/Annotations/DocLexerTest.php]: fixed

**Pull request closed without merging**